pkg/templates/notificationtemplates/notification_generator.go

```diff
@@ -22,30 +30,40 @@
 }
 
 // GetNotificationTemplate returns a notification subject, body and a boolean
-// indicating whether or not a template was found. Otherwise, an error will be returned
-func GetNotificationTemplate(name string) (*NotificationTemplate, bool, error) {
-	templates, err := loadTemplates()
+// indicating whether a template was found. Otherwise, an error will be returned
+func GetNotificationTemplate(name string, notificationEnvironment string) (*NotificationTemplate, bool, error) {
+	templates, err := loadTemplates(notificationEnvironment)
```

Comment:
I was wondering if we could simplify this and just try and read the "subject" and "notification" templates directly:

```
    subject, err := deploy.Files.ReadFile("templates/notificationtemplates/" + notificationEnvironment + "/" + name + "/subject.txt")
	if err != nil {
		return nil, false, err
	}
	content, err := deploy.Files.ReadFile("templates/notificationtemplates/" + notificationEnvironment + "/" + name + "/notification.html")
	if err != nil {
		return nil, false, err
	}

	notificationTemplate := make(map[string]NotificationTemplate)
	notificationTemplate[name] = NotificationTemplate{
		Name:    name,
		Subject: string(subject),
		Content: string(content),
	}
	
	return &notificationTemplate, true, nil
```
and we can maybe drop all the logic that loops over the FS, checks for the existance of those 2 files and uses the directory name as name of the template. If I'm not missing anything , this may replace the usage of `templatesForAssets`, `getAllFilenames`, `getTemplatesFroNotificationEnvironment`, `loadTemplates`.

NOTE: this is just sample code, it may contain errors, plus the concatenation logic can be done better, and the constants can be defined somewhere else to avoid duplication.

Reply:
I like the idea, its more straight-forward for sure! I would still keep loading the templates in a map, so that we don't have to read from the file every time to get the template. Reading from the file only once and reading from the map thereafter should be faster. 
About the way templates are read and loaded, I agree that the current implementation is not the best and it depends on the length of the segmented file path. While I think reading the html and subject directly as you have suggested is a lot easier to understand, it is still dependent on the directory structure. In short, it is not completely flexible - and if there is any change in the folder structure the flow would break and the code would need to be updated. I'll update the dependency on structure it in the comments above the function.

If there is any other way anybody can think of that makes it completely flexible, I'd be happy to incorporate it.

Reply:
Keeping the map should not be a problem , even if I think (not 100% sure), that `embed.FS.ReadFile` is quite efficient since it may have the data already loaded in memory inside the global variable you define for the filesystem.

Regarding the flexibility, as we said, the `templatesForAssets` function will still need to be updated in case we change folder structure, since it uses indexes of the splitted path to get `directoryName` and `filename`.

Anyway, as discussed, I don't mind keeping the current implementation in case I'm the only one that has a different proposal for that area of the code  :)